### ISSUE TITLE
Enable Testing for Matter 1.2

### ISSUE THREAD

**Issue by jvmahon** [home-assistant-libs/python-matter-server#402]
It looks like the first release candidate for Matter 1.2 was posted about a month ago ( https://github.com/project-chip/connectedhomeip/releases/tag/SVE_23_09%2Frc1 ) with further updates to the main branch. Is it possible to add a "testing" version of the Matter Server to HomeAssistant to begin testing this?

I'm hoping to begin work on implementing the Matter Mode Select device type on HomeAssistant, but am waiting for this next version before I begin that.

**Comment by marcelveldt** [comment]
Hi! We're about to bump the main branch to Matter 1.2 and then we'll release a BETA add-on of the Matter server in HA, as well as a 1.2 branch in HA core. I'll let you know once everything is in place.

**Comment by jvmahon** [comment]
Thank you. I look forward to it.

**Comment by jvmahon** [comment]
@marcelveldt I see that you did an update of the matter server to a 4.0 beta implementing Matter 1.2.  I don't see an option in HomeAssistant to update the current Matter server to the new beta 4.0. Is there something I need to do, or is this still in a "wait a bit" state and I'll eventually see the prompt automatically?  Thank you!

**Comment by marcelveldt** [comment]
Please wait for a couple of days as we are putting the blocks in place. We're going to launch the 1.2 based server as dev/beta add-on only for development. I'll ping you on monday when we're ready.

**Comment by jvmahon** [comment]
Thank you! I'll close this comment and just watch for the update.

**Comment by marcelveldt** [comment]
@jvmahon 
We've released a BETA add-on of the Matter server (as well as docker image) based on the 1.2 RC.
It is backwards compatible with the current state of the HA integration although I do not yet recommend using it for production as it might still contain unexpected issues, for now consider it for development only.

If you like to start working on the Matter Mode Select device type, you can spin up the beta matter server using the development add-on or run the docker container yourself. The development add-on repository is hosted here: https://github.com/home-assistant/addons-development

For the Home Assistant side (where the implementation of the device type happens), I've created a branch called "matter-1.2-dev-branch". Any Matter 1.2 related PR's can be targeted at that branch. Once Matter 1.2 is released, we'll merge that branch to HA dev.

Just let me know if you have any further questions and thanks in advance for helping out with adding a new device type support!

**Comment by jvmahon** [comment]
> Just let me know if you have any further questions and thanks in advance for helping out with adding a new device type support!

Unfortunately, it looks like my "real" job will keep me quite busy for the next few months. Hopefully, someone else may take this on.  

I think there is also a separate issue that may need to be addressed which is how to handle a hierarchy of endpoints which could be created if the Aggregator endpoint is present.  I think that would be needed for a complex device like this: https://www.brilliant.tech/products/brilliant-control-four-switch-panel?variant=43531135353026  particularly if you want to associate different Mode Select inputs with different ones of the switching inputs. I this has large implications on user interface and how to present a device.  Not sure if this has been given any thought, but I think this may need to be considered at an early stage before HomeAssistant gets too far down the line on the Matter implementation.  I'll try to post a bit more about this at a later date.
